platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/GenesisStateBuilder.java

```diff
@@ -71,9 +71,13 @@ public static ReservedSignedState buildGenesisState(
             @NonNull final PlatformContext platformContext,
             @NonNull final AddressBook addressBook,
             @NonNull final SoftwareVersion appVersion,
-            @NonNull final MerkleRoot stateRoot) {
+            @NonNull final MerkleStateRoot stateRoot) {
 
-        initGenesisPlatformState(platformContext, stateRoot.getWritablePlatformState(), addressBook, appVersion);
+        initGenesisPlatformState(
+                platformContext,
+                stateRoot.withMetrics(platformContext.getMetrics()).getWritablePlatformState(),
```

Comment:
Why was it necessary to change the `stateRoot` parameter type to `MerkleStateRoot`, and set metrics here? It seems unrelated to other changes, and it appears that `MerkleStateRoot` already is having metrics set on `init`